src/ansys/mapdl/core/launcher.py

```diff
@@ -707,39 +707,121 @@ def change_default_ansys_path(exe_loc):
         raise FileNotFoundError("File %s is invalid or does not exist" % exe_loc)
 
 
-def save_ansys_path(exe_loc=""):
-    """Find ANSYS path or query user"""
-    # if exe_loc.strip():
-    #     print('Cached ANSYS executable %s not found' % exe_loc)
-    # else:
-    #     print('Cached ANSYS executable not found')
-    exe_loc, ver = find_ansys()
-    if os.path.isfile(exe_loc):
-        # automatically cache this path
-        # print('Found ANSYS at %s' % exe_loc)
-        # resp = input('Use this location?  [Y/n]')
-        # if resp != 'n':
+def save_ansys_path(exe_loc=None):  # pragma: no cover
+    """Find ANSYS path or query user.
+
+    If no ``exe_loc`` argument is supplied, this function attempt
+    to obtain the MAPDL executable from (and in order):
+    - The default ansys paths (i.e. 'C:/Program Files/Ansys Inc/vXXX/ansys/bin/ansysXXX')
+    - The configuration file
+    - User input
+
+    If ``exe_loc`` is supplied, this function does some checks.
+    If successful, it will write that ``exe_loc`` into the config file.
+
+    Parameters
+    ----------
+    exe_loc : str, optional
+        Path of the MAPDL executable ('ansysXXX'), by default None
+
+    Returns
+    -------
+    str
+        Path of the MAPDL executable.
+
+    Notes
+    -----
+    The configuration file location (``config.txt``) can be found in
+    ``appdirs.user_data_dir("ansys_mapdl_core")``. For example:
+    >>> import appdirs
+    >>> import os
+    >>> print(os.path.join(appdirs.user_data_dir("ansys_mapdl_core"), "config.txt"))
+    C:/Users/gayuso/AppData/Local/ansys_mapdl_core/ansys_mapdl_core/config.txt
+
+    You can change the default ``exe_loc`` either by modifying the mentioned
+    ``config.txt`` file or by executing this function:
+    >>> from ansys.mapdl.core.launcher import save_ansys_path
+    >>> save_ansys_path('/new/path/to/executable')
```

Comment:
This has to use sphinx directives.

```suggestion
    ``appdirs.user_data_dir("ansys_mapdl_core")``. For example:
    
    .. code::

        >>> import appdirs
        >>> import os
        >>> print(os.path.join(appdirs.user_data_dir("ansys_mapdl_core"), "config.txt"))
        C:/Users/gayuso/AppData/Local/ansys_mapdl_core/ansys_mapdl_core/config.txt

    You can change the default ``exe_loc`` either by modifying the mentioned
    ``config.txt`` file or by executing this function:
    
    .. code::

       >>> from ansys.mapdl.core.launcher import save_ansys_path
       >>> save_ansys_path('/new/path/to/executable')
```

Reply:
True! I don't know what I was thinking...